owner-onboarding-server/src/ov_management/mod.rs

```diff
@@ -0,0 +1,242 @@
+        .untuple_one()
+        .and_then(delete_ov);
+        add.or(delete)
+
+}
+
+async fn add_ov(
+    count: HeaderValue,
+    content_type: HeaderValue, 
+    body: impl Buf,
+    user_data: Arc<OwnerServiceUD>,
+) -> Result<impl warp::Reply,warp::Rejection> 
+{   
+    // check for ov type
+    let content_type = content_type.to_str().unwrap();
+    let ovs= match content_type {
+        "application/x-pem-file" => OwnershipVoucher::many_from_pem(body.chunk()),
+        "application/cbor" => OwnershipVoucher::deserialize_many_from_reader(body.chunk()),
+        _ => {
+            let  e =  ErrorMsg::generate(ErrorCode::RequireContentTypeHeader, ACCEPTED_CONTENT_TYPE);
+            return Ok(warp::reply::with_status(e, StatusCode::BAD_REQUEST));
+        }
+    };
+
+    // parse-error
+    if ovs.is_err() {
+        let  e =  ErrorMsg::generate(ErrorCode::ParseError, ovs.unwrap_err().to_string());
+        return Ok(warp::reply::with_status(e, StatusCode::BAD_REQUEST));
+    }
+
+    // check-empty
+    let ovs = ovs.unwrap();
+    if ovs.is_empty() {
+        let  e =  ErrorMsg::generate(ErrorCode::ParseError, "Ownership voucher not found!!");
+        return Ok(warp::reply::with_status(e, StatusCode::BAD_REQUEST));
+    }
+
+    // check if X-Number-Of-Vouchers present
+    let count = count.to_str().unwrap().parse::<usize>();
+    if count.is_err() {
+        let  e =  ErrorMsg::generate(ErrorCode::RequireVoucherCountHeader, "X-Number-Of-Vouchers: Should be postive Interger");
+        return Ok(warp::reply::with_status(e, StatusCode::BAD_REQUEST));
+    }
+    
+
+    // check if count matches
+    if ovs.len() != count.unwrap() {
+        let  e =  ErrorMsg::generate(ErrorCode::InvalidNumberOfVouchers, ovs.len());
+        return Ok(warp::reply::with_status(e, StatusCode::BAD_REQUEST));
+    }
+    let mut unowned_guid = Vec::new();
+    for ov in ovs.iter() {
+        if !validate_ov(ov,&user_data.owner_key){           
+            unowned_guid.push(ov.header().guid().to_string());
+        
+        }
+    }
+    
+    if !unowned_guid.is_empty() {
+        let  e =  ErrorMsg::generate(ErrorCode::UnwonedVoucher, unowned_guid);
+        return Ok(warp::reply::with_status(e, StatusCode::BAD_REQUEST));
+    }
+    
+    let mut owned_guid = Vec::new();
+    for ov in ovs.iter() {
+            let o = ov.header().guid().to_string();
+            owned_guid.push(o);
+            let _ = user_data
+                .ownership_voucher_store
+                .store_data(ov.clone().header().guid().to_owned(), ov.clone())
+                .await;
+        }
+
+    let uuid = warp::reply::json(&owned_guid);
+    Ok(warp::reply::with_status(uuid, StatusCode::CREATED))
+
+}
+
+async fn delete_ov(
+    body: impl Buf,
+    user_data: Arc<OwnerServiceUD>,
+) -> Result<Box<dyn warp::reply::Reply>,warp::Rejection> {
+
+    let u = String::from_utf8(body.chunk().to_vec()).expect("Found invalid UTF-8");
+    let uuids_to_delete:Vec<String> = u
```

Comment:
This looks like manual json parsing. That is a very bad idea.
Please just use the warp json features for parsing them, and then also use that to parse to `Guid`.